packages/hint/src/lib/cli/analyze.ts

```diff
@@ -104,46 +104,64 @@ const pruneUserConfig = (userConfig: UserConfig) => {
     };
 };
 
-/** Ask user if he wants to activate the telemetry or not. */
-const askForTelemetryConfirmation = async (userConfig: UserConfig) => {
-    if (appInsights.isConfigured()) {
-        return;
-    }
+const askForTelemetryConfirmation = async () => {
+    showTelemetryMessage();
 
-    if (isCI) {
-        if (!appInsights.isConfigured()) {
-            showCITelemetryMessage();
-        }
+    const message = `Do you want to opt-in?`;
 
-        return;
-    }
-
-    const alreadyRun: boolean = configStore.get(configStoreKey);
+    debug(`Prompting telemetry permission.`);
 
-    if (!alreadyRun) { /* This is the first time, don't ask anything. */
-        configStore.set(configStoreKey, true);
+    const telemetryEnabled = await askQuestion(message);
 
-        return;
+    if (telemetryEnabled) {
+        appInsights.enable();
+    } else {
+        appInsights.disable();
     }
 
-    showTelemetryMessage();
+    return telemetryEnabled;
+};
 
-    const message: string = `Do you want to opt-in?`;
+/** Ask user if he wants to activate the telemetry or not. */
+const sendTelemetryIfEnabled = async (userConfig: UserConfig) => {
+    const telemetryConfigured = appInsights.isConfigured();
+    let telemetryEnabled = appInsights.isEnabled();
+    const alreadyRun: boolean = configStore.get(alreadyRunKey);
 
-    debug(`Prompting telemetry permission.`);
+    if (!alreadyRun) {
+        configStore.set(alreadyRunKey, true);
+    }
 
-    const confirm: boolean = await askQuestion(message);
+    if (!telemetryConfigured) {
+        if (isCI) {
+            showCITelemetryMessage();
+            telemetryEnabled = false;
+        } else if (alreadyRun) {
+            /* Only prompt the user about opt-intelemetry if they have run webhint before. */
+            telemetryEnabled = await askForTelemetryConfirmation();
+        }
+    }
 
-    if (confirm) {
-        appInsights.enable();
+    if (!telemetryEnabled) {
+        return;
+    }
 
-        appInsights.trackEvent('SecondRun');
-        appInsights.trackEvent('analyze', pruneUserConfig(userConfig));
+    if (!alreadyRun && !isCI) {
+        appInsights.trackEvent('FirstRun');
+    }
 
-        return;
+    /**
+     * No need to check for CI as state is not persisted and
+     * `alreadyRun` will always be `false`.
+     */
+    if (alreadyRun) {
+        appInsights.trackEvent('SecondRun');
     }
 
-    appInsights.disable();
+    const prunedUserConfig = pruneUserConfig(userConfig);
+    const payload = { ...{ ci: isCI }, ...prunedUserConfig };
```

Comment:
No need to create an intermediate object here.

```suggestion
    const payload = { ci: isCI, ...prunedUserConfig };
```